src/terms/ewald.jl

```diff
@@ -33,35 +33,52 @@ function energy_ewald(model::Model{T}; kwargs...) where {T}
+# of the energy term is computed in the reciprocal space and the second (singular) one in
+# the real-space.
+# `η` is an arbitrary parameter that enables to balance the computation of those to parts.
+# By default, we choose it to have a slight bias towards the reciprocal summation.
+function default_η(lattice::AbstractArray{T}) where {T}
+    any(iszero.(eachcol(lattice))) && return  # We won't compute anything
+    recip_lattice = compute_recip_lattice(lattice)
+    sqrt(sqrt(T(1.69) * norm(recip_lattice ./ 2T(π)) / norm(lattice))) / 2
+end
+
 # This could be merged with Pairwise, but its use of `atom_types` would slow down this
 # computationally intensive Ewald sums. So we leave it as it for now.
 """
-Compute the electrostatic interaction energy per unit cell between point charges in
-a uniform background of compensating charge to yield net neutrality.`lattice` should
-contain the lattice vectors as columns. `charges` and `positions` are the point charges and
-their positions (as an array of arrays) in fractional coordinates. If `forces` is not
-nothing, minus the derivatives of the energy with respect to `positions` is computed.
+Computes the local energy and forces on the atoms of the reference unit cell 0, for an
+infinite array of atoms at positions r_{iR} = positions[i] + R + ph_disp[i]*e^{iq·R} in
+a uniform background of compensating charge to yield net neutrality.
+`lattice` should contain the lattice vectors as columns. `charges` and `positions` are the
+point charges and their positions (as an array of arrays) in fractional coordinates. If
+`forces` is not nothing, minus the derivatives of the energy with respect to `positions` is
+computed.
+
+`q` is the phonon `q`-point (`Vec3`), and `ph_disp` a list of `Vec3` displacements to
+compute the Fourier transform of the force constant matrix.
 
 For now this function returns zero energy and force on non-3D systems. Use a pairwise
 potential term if you want to customise this treatment.
 """
-function energy_ewald(lattice::AbstractArray{T}, charges, positions;
-                      η=nothing, forces=nothing) where {T}
+function energy_ewald(lattice::AbstractArray{T}, charges, positions; η=default_η(lattice),
+                      forces=nothing, q=nothing, ph_disp=nothing) where {T}
     # TODO should something more clever be done here? For now
     # we assume that we are not interested in the Ewald
     # energy of non-3D systems
     any(iszero.(eachcol(lattice))) && return zero(T)
 
     recip_lattice = compute_recip_lattice(lattice)
     @assert length(charges) == length(positions)
-    if isnothing(η)
-        # Balance between reciprocal summation and real-space summation
-        # with a slight bias towards reciprocal summation
-        η = sqrt(sqrt(T(1.69) * norm(recip_lattice ./ 2T(π)) / norm(lattice))) / 2
+    if !isnothing(ph_disp)
+        @assert !isnothing(q) && !isnothing(forces)
```

Comment:
it should be possible to have ph_disp non-nothing but not compute the forces (in which case we just compute the local energy)